addons/stock/models/stock_move.py

```diff
@@ -2116,23 +2116,24 @@ def _trigger_scheduler(self):
         if not self or self.env['ir.config_parameter'].sudo().get_param('stock.no_auto_scheduler'):
             return
 
-        orderpoints_by_company = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
-        orderpoints_context_by_company = defaultdict(dict)
+        all_orderpoints = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
+        for orderpoint in self.env['stock.warehouse.orderpoint'].search([
+            ('product_id', 'in', self.product_id.ids),
+            ('trigger', '=', 'auto'),
+            ('company_id', 'in', self.company_id.ids),
+        ]):
+            all_orderpoints[(orderpoint.company_id, orderpoint.product_id)] |= orderpoint
+
+        orderpoints = defaultdict(lambda: defaultdict(list))
         for move in self:
-            orderpoint = self.env['stock.warehouse.orderpoint'].search([
-                ('product_id', '=', move.product_id.id),
-                ('trigger', '=', 'auto'),
-                ('location_id', 'parent_of', move.location_id.id),
-                ('company_id', '=', move.company_id.id),
-                '!', ('location_id', 'parent_of', move.location_dest_id.id),
-            ], limit=1)
+            orderpoint = all_orderpoints[(move.company_id, move.product_id)].filtered(lambda o: str(o.location_id.id) in move.location_id.parent_path and not str(o.location_id.id) in move.location_dest_id.parent_path)[:1]
```

Comment:
I created an helper function in stock.location `_child_of` maybe you can use it to make the code a bit easier to read

Reply:
usefull, thank you